src/peft/tuners/lora/model.py

```diff
@@ -431,12 +438,20 @@ def _unload_and_optionally_merge(self, merge=True, progressbar: bool = False, sa
                         new_module = torch.nn.Linear(target.in_features, target.out_features, bias=bias)
                 if merge:
                     target.merge(safe_merge=safe_merge)
-                self._replace_module(parent, target_name, new_module, target)
+
+                if hasattr(module, "_hf_hook") and isinstance(module._hf_hook, AlignDevicesHook):
+                    module._hf_hook.post_forward(module, torch.tensor([]))
```

Comment:
When I tested this with the bloomz model, this line was never reached.

Reply:
Note: c.f. the discussion hook problems for smaller modules with LoRA parameters